detection_rules/integrations.py

```diff
@@ -187,54 +190,72 @@ def find_latest_compatible_version(package: str, integration: str,
         if not compatible_versions:
             raise ValueError(f"Manifest for {package}:{integration} version {version} is missing compatible versions")
 
-        highest_compatible_version = max(compatible_versions, key=lambda x: Version(x))
+        highest_compatible_version = Version.parse(max(compatible_versions,
+                                                       key=lambda x: Version.parse(x)))
 
-        if Version(highest_compatible_version) > Version(rule_stack_version):
+        if highest_compatible_version > rule_stack_version:
             # generate notice message that a later integration version is available
             integration = f" {integration.strip()}" if integration else ""
 
             notice = (f"There is a new integration {package}{integration} version {version} available!",
                       f"Update the rule min_stack version from {rule_stack_version} to "
                       f"{highest_compatible_version} if using new features in this latest version.")
 
-        if int(highest_compatible_version[0]) == int(rule_stack_version[0]):
+        if highest_compatible_version.major == rule_stack_version.major:
             return version, notice
 
         else:
             # Check for rules that cross majors
             for compatible_version in compatible_versions:
-                if Version(compatible_version) <= Version(rule_stack_version):
+                if Version.parse(compatible_version) <= rule_stack_version:
                     return version, notice
 
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
-def get_integration_manifests(integration: str) -> list:
+def get_integration_manifests(integration: str, prerelease: str, kibana_version: str) -> list:
     """Iterates over specified integrations from package-storage and combines manifests per version."""
     epr_search_url = "https://epr.elastic.co/search"
+    if not prerelease:
+        prerelease = "false"
 
     # link for search parameters - https://github.com/elastic/package-registry
-    epr_search_parameters = {"package": f"{integration}", "prerelease": "false",
+    epr_search_parameters = {"package": f"{integration}", "prerelease": prerelease,
                              "all": "true", "include_policy_templates": "true"}
+    if kibana_version:
+        epr_search_parameters["kibana.version"] = kibana_version
     epr_search_response = requests.get(epr_search_url, params=epr_search_parameters)
```

Comment:
Might want to add a timeout param to the `request.get`

Reply:
Added 10 as default.